pkg/asset/machines/openstack/machinesets.go

```diff
@@ -32,23 +32,19 @@ func MachineSets(clusterID string, config *types.InstallConfig, pool *types.Mach
 		total = int32(*pool.Replicas)
 	}
 
-	numOfAZs := int32(len(mpool.Zones))
-	// TODO(flaper87): Add support for availability zones
 	var machinesets []*clusterapi.MachineSet
 
-	for idx, az := range mpool.Zones {
-		replicas := int32(total / numOfAZs)
-		if int32(idx) < total%numOfAZs {
-			replicas++
-		}
+	for _, az := range mpool.Zones {
 		provider, err := generateProvider(clusterID, platform, mpool, osImage, az, role, userDataSecret, trunkSupport)
 		if err != nil {
 			return nil, err
 		}
 
-		// TODO(flaper87): Implement AZ support sometime soon
-		//name := fmt.Sprintf("%s-%s-%s", clustername, pool.Name, az)
-		name := fmt.Sprintf("%s-%s", clusterID, pool.Name)
+		name := fmt.Sprintf("%s-%s-%s", clusterID, pool.Name, az)
+		if az == "" {
+			name = fmt.Sprintf("%s-%s", clusterID, pool.Name)
+		}
```

Comment:
nit: you could avoid reflection with string concatenation (just a matter of taste!)

```Go
name := clusterID + "-" + pool.Name
if az != "" {
	name += "-" + az
}
```

Reply:
yep, it looks better!